capplets/windows/mate-window-properties.c

```diff
@@ -203,17 +198,44 @@ double_click_titlebar_changed_callback (GtkWidget *optionmenu,
                          gtk_combo_box_get_active (GTK_COMBO_BOX (optionmenu)));
 }
 
+static void
+on_titlebar_button_layout_entry_activate (GtkWidget *entry,
+                                          void      *data)
+{
+    g_settings_set_string (marco_settings, MARCO_BUTTON_LAYOUT_KEY, gtk_entry_get_text(GTK_ENTRY(entry)));
```

Comment:
It is dangerous not to check what ```gtk_entry_get_text(GTK_ENTRY(entry)）```returns strings. If the user does not enter anything, or if the user enters the wrong content, this will cause ```marco``` to restart repeatedly.